pkg/symbolizer/symbolizer.go

```diff
@@ -89,13 +96,19 @@ func NewSymbolizer() (*Symbolizer, error) {
 	if !ok {
 		return nil, fmt.Errorf("reading inode of %s/1/ns/pid", host.HostProcFs)
 	}
+	goReSyms, err := parseGoReSym([]byte(opts.GoReSymSpec))
+	if err != nil {
+		return nil, fmt.Errorf("parsing goresym: %w", err)
+	}
 
 	s := &Symbolizer{
 		symbolTables:    make(map[exeKey]*symbolTable),
 		hostProcFsPidNs: uint32(pid1PidNsStat.Ino),
 		pruneTickerTime: pruneTickerTime,
 		symbolTableTTL:  symbolTableTTL,
 		exit:            make(chan struct{}),
+
+		goReSyms: goReSyms,
```

Comment:
What about making this logic more modular? Like having a common interface (basically `Resolve(task Task, addresses []uint64) ([]string, error)`) and then implementing it in different resolvers (existing resolver, GoReSym added on this PR, etc).